tools/export/cdt/__init__.py

```diff
@@ -1,18 +1,63 @@
-import re
+"""
+mbed SDK
+Copyright (c) 2016-2019 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
 
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import re
+import os
+import json
+from collections import namedtuple
+from tools.targets import TARGET_MAP
 from os.path import join, exists
 from os import makedirs, remove
 import shutil
+from copy import deepcopy
 
 from tools.export.makefile import Makefile, GccArm, Armc5, IAR
 
+_eclipse_defs = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'cdt_definitions.json')
+
+with open(_eclipse_defs, 'r') as f:
+    _CONFIGS_OPTIONS = json.load(f)
+
+supported_launches = ['debug', 'program', 'erase']
+
 class Eclipse(Makefile):
     """Generic Eclipse project. Intended to be subclassed by classes that
     specify a type of Makefile.
     """
+    def get_target_config(self, ctx, configuration):
+        """Retrieve info from cdt_definitions.json"""
+        tgt = deepcopy(TARGET_MAP[self.target])
```

Comment:
I don't think you need this deepcopy. Further, I think you only use `tgt.name` which is always the same as `self.target`.